### PR TITLE
CORE-837 User can view all teams or my teams

### PR DESCRIPTION
**Pull request by ash3rz**
This also adds teams to the global search. Creating/editing a team will be the next PR, so clicking on a team doesn't do anything right now.

Desktop view:
image

By default All Teams is selected so new users don't see a blank view:
image

Mobile view:
image

Search dropdown list:
image

Search detailed view:
image

### CODE REVIEW

File: src/components/sharing/UserTable.js [hunk 1]
```diff
@@ -91,59 +81,8 @@ function UserTable(props) {
...
-                                        {cell.render("Cell")}
-                                    </TableCell>
-                                );
-                            })}
-                        </TableRow>
-                    );
-                })}
-            </TableBody>
-        </Table>
-    );
```
Comment (ash3rz) [inline src/components/sharing/UserTable.js:146]: All of this ☝️ was pretty much copy-pasted into `BasicTable`, so other components can hopefully reuse it.  The component name can obviously be changed.

File: src/components/search/GlobalSearchField.js [hunk 2]
```diff
@@ -325,6 +334,50 @@ function AnalysesSearchOption(props) {
...
+    }
+
+    const href = `/${NavigationConstants.TEAMS}`;
+    const as = `/${NavigationConstants.TEAMS}`;
+    return (
+        <Link href={href} as={as} passHref>
+            <SearchOption
+                primary={selectedOption.display_extension}
+                secondary={selectedOption.description}
+                icon={<TeamIcon color={theme.palette.info.primary} />}
```
Comment (sri2k1us) [inline src/components/search/GlobalSearchField.js:373]: I don't think this works. You will have to use style instead.
`style={{theme.palette.info.main}}`

File: src/components/teams/Icons.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import React from "react";
+import { Group, GroupAdd } from "@material-ui/icons";
+
+export function TeamIcon(props) {
+    return <Group {...props} />;
+}
+
+export function AddTeamIcon(props) {
+    return <GroupAdd {...props} />;
+}
```
Comment (sri2k1us) [inline src/components/teams/Icons.js:16]: 👍🏽

File: src/components/utils/BasicTable.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        </Table>
+    );
+}
+
+BasicTable.propTypes = {
+    columns: PropTypes.array.isRequired,
+    data: PropTypes.array.isRequired,
+};
+
+export default BasicTable;
```
Comment (sri2k1us) [inline src/components/utils/BasicTable.js:96]: Awesome 🎉 . One of my wishlist 👍🏽

File: src/components/utils/DELink.js [hunk 5]
```diff
@@ -34,7 +34,7 @@ const DELink = React.forwardRef((props, ref) => {
             href={href}
             onClick={onClick}
             ref={ref}
-            className={classes.dataLink}
+            className={classes.link}
```
Comment (sri2k1us) [inline src/components/utils/DELink.js:37]: 👀

**Review by sri2k1us — APPROVED** [review]
LGTM. I only had minor comments. Also it would be good to add this help text in the page somewhere.
image

File: src/components/teams/Icons.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @author aramsey
+ *
+ * A single place to define what the icons for teams will be throughout the app
```
Comment (psarando) [inline src/components/teams/Icons.js:4]: 👍

File: src/components/teams/Listing.js [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+    const { t } = useTranslation(["teams", "search"]);
+    const [data, setData] = useState([]);
+    const [userProfile] = useUserProfile();
+
+    const tableId = build(parentId, ids.TEAMS.TABLE);
+    const TEAM_COLUMNS = Columns(t);
+    const isSearchPage = !!searchTerm;
+
+    const columns = useMemo(() => {
```
Comment (psarando) [inline src/components/teams/Listing.js:64]: 👍

**Review by psarando — APPROVED** [review]
Very tidy ✨ which made this a quick review 🎉 
:shipit: 👍

**Comment by ash3rz** [comment]
Thanks for the reviews!

**Pull request merged**